### PR TITLE
convco: init at 0.3.2

### PR DESCRIPTION
**Pull request by Hoverbear**
###### Motivation for this change

Introduce the `convco` tool from https://github.com/convco/convco.

Example usage, while in a conventional commit respecting git repo like [Mayastor](https://github.com/openebs/Mayastor) or [Junctor](https://github.com/Hoverbear-Consulting/junctor) or [Vector](https://github.com/timberio/vector):

```bash
convco changelog > CHANGELOG.md
convco commit --refactor
convco version
```

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -3699,6 +3699,12 @@
     githubId = 362833;
     name = "Hongchang Wu";
   };
+  hoverbear = {
+    email = "operator+nix@hoverbear.org";
+    github = "hoverbear";
+    githubId = "130903";
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix]:
```suggestion
    githubId = 130903;
```

Please put this into a separate commit with the message ``maintainers: add hoverbear``.

File: pkgs/development/tools/convco/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{ stdenv, fetchFromGitHub, openssl, perl, pkg-config, rustPlatform
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "convco";
+  version = "v0.3.2";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/convco/default.nix:6]:
```suggestion
  version = "0.3.2";
```

File: pkgs/development/tools/convco/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+  src = fetchFromGitHub {
+    owner = "convco";
+    repo = pname;
+    rev = version;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/convco/default.nix:11]:
```suggestion
    rev = "v${version}";
```

File: pkgs/development/tools/convco/default.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    sha256 = "0fqq6irbq1aikhhw08gc9kp0vbk2aminfbvwdlm58cvywyq91bn4";
+  };
+
+  cargoSha256 = "0jklhv6rxlw0r10yhiiih36x9yf6f1i1255lcz1pfvzw8za8qmfd";
+
+  nativeBuildInputs = [ openssl perl pkg-config ];
+
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/development/tools/convco/default.nix:19]:
```suggestion
  meta = with lib; {
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by Hoverbear** [comment]
@SuperSandro2000 Gosh, thanks for the fast review. :) I've resolved those issues and will be mindful of them in the future!

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110375` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>convco: log <a href="https://termbin.com/c4c3">https://termbin.com/c4c3</a></li>
  </ul>
</details>

File: pkgs/development/tools/convco/default.nix [hunk 5]
```diff
@@ -0,0 +1,25 @@
+{ stdenv, lib, fetchFromGitHub, openssl, perl, pkg-config, rustPlatform
```
Comment (cole-h) [inline pkgs/development/tools/convco/default.nix:1]: Since `stdenv` is unused, we can get rid of it!

Additionally, we should move `rustPlatform` higher in the order as a result of the recently-merged https://github.com/NixOS/nixpkgs/pull/110060 (which boils down to: "arguments should be listed in the order they are used, with the exception of `lib` which should always go first").

```suggestion
{ lib, rustPlatform, fetchFromGitHub, openssl, perl, pkg-config
```

**Comment by Hoverbear** [comment]
Hmmm it looks like there is an iconv error:

```
  cargo:warning=In file included from libgit2/src/commit_list.c:12:
  cargo:warning=In file included from libgit2/src/odb.h:19:
  cargo:warning=In file included from libgit2/src/filter.h:12:
  cargo:warning=In file included from libgit2/src/attr_file.h:17:
  cargo:warning=In file included from libgit2/src/futils.h:14:
  cargo:warning=libgit2/src/path.h:430:10: fatal error: 'iconv.h' file not found
  cargo:warning=#include <iconv.h>
  cargo:warning=         ^~~~~~~~~
  cargo:warning=In file included from libgit2/src/merge_driver.c:12:
  cargo:warning=In file included from libgit2/src/merge.h:15:
  cargo:warning=In file included from libgit2/src/iterator.h:15:
  cargo:warning=In file included from libgit2/src/ignore.h:12:
  cargo:warning=In file included from libgit2/src/repository.h:24:
  cargo:warning=In file included from libgit2/src/attrcache.h:12:
  cargo:warning=In file included from libgit2/src/attr_file.h:17:
  cargo:warning=In file included from libgit2/src/futils.h:14:
  cargo:warning=libgit2/src/path.h:430:10: fatal error: 'iconv.h' file not found
  cargo:warning=#include <iconv.h>
  cargo:warning=         ^~~~~~~~~
  cargo:warning=1 error generated.
  exit code: 1
  ```

File: pkgs/development/tools/convco/default.nix (hunk 5 above)
Comment (Hoverbear) [reply pkgs/development/tools/convco/default.nix:1]: Stdenv needs to be used for `isDarwin` sadly!

Comment (cole-h) [reply pkgs/development/tools/convco/default.nix:1]: Maybe it's just me, but I don't see any `isDarwin` usage ;)

**Review by cole-h — APPROVED** [review]
Diff LGTM and runs fine. Using the instructions provided in the PR body, I was able to generate a `CHANGELOG.md` for Vector. Cool tool!

---

Result of `nixpkgs-review pr 110375` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>convco</li>
  </ul>
</details>

File: pkgs/development/tools/convco/default.nix [hunk 6]
```diff
@@ -0,0 +1,26 @@
+{ lib, rustPlatform, fetchFromGitHub, stdenv, libiconv, openssl, perl, pkg-config,
```
Comment (SuperSandro2000) [inline pkgs/development/tools/convco/default.nix:1]:
```suggestion
{ lib, rustPlatform, fetchFromGitHub, stdenv, libiconv, openssl, perl, pkg-config
```

File: pkgs/development/tools/convco/default.nix [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+}:
```
Comment (SuperSandro2000) [inline pkgs/development/tools/convco/default.nix:2]:
```suggestion
{ lib, rustPlatform, fetchFromGitHub, stdenv, libiconv, openssl, perl, pkg-config }:
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110375` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>convco: log <a href="https://termbin.com/o0as">https://termbin.com/o0as</a></li>
  </ul>
</details>

darwin patch
```diff
diff --git a/pkgs/development/tools/convco/default.nix b/pkgs/development/tools/convco/default.nix
index 2e1987e72d6..b3071f8f387 100644
--- a/pkgs/development/tools/convco/default.nix
+++ b/pkgs/development/tools/convco/default.nix
@@ -1,5 +1,5 @@
-{ lib, rustPlatform, fetchFromGitHub, stdenv, libiconv, openssl, perl, pkg-config,
-}:
+{ lib, rustPlatform, fetchFromGitHub, stdenv, openssl, perl, pkg-config
+, libiconv, Security }:
 
 rustPlatform.buildRustPackage rec {
   pname = "convco";
@@ -14,9 +14,10 @@ rustPlatform.buildRustPackage rec {
 
   cargoSha256 = "073sfv42fbl8rjm3dih1ghs9vq75mjshp66zdzdan2dmmrnw5m9z";
 
-  buildInputs = lib.optionals stdenv.isDarwin [ libiconv ];
   nativeBuildInputs = [ openssl perl pkg-config ];
 
+  buildInputs = lib.optionals stdenv.isDarwin [ libiconv Security ];
+
   meta = with lib; {
     description = "A Conventional commit cli";
     homepage = "https://github.com/convco/convco";
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index e903d966f58..46a9d0bdcac 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -10695,7 +10695,9 @@ in
 
   crate2nix = callPackage ../development/tools/rust/crate2nix { };
 
-  convco = callPackage ../development/tools/convco { };
+  convco = callPackage ../development/tools/convco {
+    inherit (darwin.apple_sdk.frameworks) Security;
+  };
 
   maturin = callPackage ../development/tools/rust/maturin { };
   inherit (rustPackages) rls;
```

**Comment by Hoverbear** [comment]
Thank you -- Darwin patch applied.

File: pkgs/top-level/all-packages.nix [hunk 8]
```diff
@@ -10691,6 +10691,9 @@ in
   };
 
   crate2nix = callPackage ../development/tools/rust/crate2nix { };
+  convco = callPackage ../development/tools/convco {
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:10694]:
```suggestion
  crate2nix = callPackage ../development/tools/rust/crate2nix { };

  convco = callPackage ../development/tools/convco {
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110375` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>convco</li>
</ul>
</details>
